src/Draco.Compiler/Internal/Documentation/DocumentationSection.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Draco.Compiler.Internal.Documentation;
+
+/// <summary>
+/// Represents a section of the documentation.
+/// </summary>
+internal sealed class DocumentationSection
+{
+    public string? Name { get; }
+    public ImmutableArray<DocumentationElement> Elements { get; }
+
+    public SectionKind Kind => this.kind ?? SectionKind.Other;
+    private SectionKind? kind;
```

Comment:
No need for this field, just assign Other if the name didn't match anything known